### PR TITLE
Fixed error handling

### PR DESCRIPTION
**Pull request by gavraz**

### CODE REVIEW

File: activation/atxdb.go [hunk 1]
```diff
@@ -285,17 +285,18 @@ func (db *ActivationDb) GetAtx(id types.AtxId) (*types.ActivationTx, error) {
...
+	if err != nil { // means there is no such identity
+		log.Error("IsIdentityActive erred while getting identity err=%v", err)
+		return false, nil
 	}
 	ids, err := db.GetNodeAtxIds(nodeId)
 	if err != nil {
 		return false, err
 	}
-	if len(ids) == 0 {
+	if len(ids) == 0 { // GetIdentity succeeded but no ATXs, this is a fatal error
```
Comment (zalmen) [inline activation/atxdb.go:299]: add more context to the error - "no active IDs for known node".

File: hare/algorithm.go [hunk 2]
```diff
@@ -55,19 +56,26 @@ type State struct {
...
+	IsIdentityActive(edId string, layer types.LayerID) (bool, error)
 }
 
 type Msg struct {
 	*pb.HareMessage
 	PubKey []byte
 }
 
-type mockStateQuerier struct {
+// TODO: move to unit test
```
Comment (zalmen) [inline hare/algorithm.go:67]: move it as part of this PR (bet you whatever you want that you knew that I'll write this comment ;)

**Review by zalmen — CHANGES_REQUESTED** [review]

File: activation/atxdb.go (hunk 1 above)
Comment (gavraz) [reply activation/atxdb.go:299]: Fixed

File: hare/algorithm.go (hunk 2 above)
Comment (gavraz) [reply hare/algorithm.go:67]: You can't use anything from unit test files (it is not part of the compiled executable)

**Review by zalmen — APPROVED** [review]

**Pull request merged**